### PR TITLE
Data Processing functions

### PR DESCRIPTION
**Pull request by sonaalll**
Added functions to clean the data before processing it.

### CODE REVIEW

File: climate_scanner/coarse_article_classifier/data_utils.py [hunk 1]
```diff
@@ -26,4 +43,82 @@ def data_cleaning(self, text):
         while ensuring least possible loss of information
         """
         pass
+
+    def _remove_hyperlinks(self, text):
+        """ This function removes hyperlinks from texts
+            inputs:
+            - text """ 
+        text = text.apply(lambda x: re.sub(r"http\S|www\.\S++", " ", x))
```
Comment (Eliotfrazier) [inline climate_scanner/coarse_article_classifier/data_utils.py:51]: Are theses apply(lambda... wrappers necessary here? I think you should be able to just apply the functions directly.

**Review by Eliotfrazier — CHANGES_REQUESTED** [review]
Looks reasonable to me, but check my comment regarding function application.  :)

**Comment by Eliotfrazier** [comment]
@sonaalll did this comment make sense?

File: climate_scanner/coarse_article_classifier/data_utils.py (hunk 1 above)
Comment (kandalavaishnavi) [reply climate_scanner/coarse_article_classifier/data_utils.py]: in fact, you can replace all of them and use just one apply function. Please refer to the data_processing step in the trends_innovation_classifier.

File: climate_scanner/coarse_article_classifier/data_utils.py [hunk 2]
```diff
@@ -13,11 +21,31 @@ def pre_processing(self, text):
...
+        text = text.apply(lambda x: unicodedata.normalize("NFD", x).encode('ascii', 'ignore').decode("utf-8"))
+        # clean end-of-line tabs
+        text = text.str.lower()
+        text = text.apply(lambda x: x.replace("\n", " "))
+        text = text.apply(lambda x: x.replace("\r", " "))
+        text = text.apply(lambda x: x.replace("\t", " "))
+        # clean punctuations
+        text = text.apply(lambda x: re.sub(r'[^\w\s]', ' ', x))
+        # remove stopwords
+        text = text.apply(lambda x: self._remove_stop_words(x))
```
Comment (kandalavaishnavi) [inline climate_scanner/coarse_article_classifier/data_utils.py:44]: @sonaalll  - when doing contextual embeddings especially, we shouldn't be removing stop words - we might lose some important tokens which will create a completely different embedding. We can remove extra spaces, HTML, Unicode characters. I am also thinking about the numbers - @Eliotfrazier  - I think we can keep them as well! Let us know your thoughts!

**Review by kandalavaishnavi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**